### PR TITLE
[docs] Improve custom TypeScript functions docs

### PR DESCRIPTION
**Pull request by szdziedzic**
# Why

Improvements based on this feedback https://exponent-internal.slack.com/archives/C06EFBQK3B7/p1708739017961519?thread_ts=1708027959.160569&cid=C06EFBQK3B7

# How

Don't use `npm install` because it might not work for every project. Use `eas/install_node_modules` instead.

Link to the resources helping you to create your first workflow and plug it into `eas.json`

# Test Plan

Preview

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/custom-builds/functions.mdx [hunk 1]
```diff
@@ -61,13 +61,9 @@ export default myFunction;
...
 
-## Compiling the function and connect it to the config file
+## Compile the function
 
-After generating the function, you will want to compile it and connect it to your custom build YAML configuration.
-
-### Compile the function
-
-Functions must be compiled to a single JavaScript file that can be run without installing any Node Module dependencies. The default `build` script for generated functions uses the [ncc](https://github.com/vercel/ncc) to compile your function into a single file with all its dependencies. If you don't have the `ncc` installed globally on your machine, run `npm install -g @vercel/ncc` to install it. Next, run the build script in the  **.eas/build/myFunction** directory:
+Functions must be compiled to a single JavaScript file that can be run without installing any Node Module dependencies. The default `build` script for generated functions uses the [ncc](https://github.com/vercel/ncc) to compile your function into a single file with all its dependencies. If you don't have the `ncc` installed globally on your machine, run `npm install -g @vercel/ncc` to install it. Next, run the build script in the **.eas/build/myFunction** directory:
```
Comment (sjchmiela) [inline docs/pages/custom-builds/functions.mdx:66]:
```suggestion
Functions must be compiled to a single JavaScript file that can be run without installing any dependencies. The default `build` script for generated functions uses [ncc](https://github.com/vercel/ncc) to compile your function into a single file with all its dependencies. If you don't have the `ncc` installed globally on your machine, run `npm install -g @vercel/ncc` to install it. Next, run the build script in the **.eas/build/myFunction** directory:
```

File: docs/pages/custom-builds/functions.mdx [hunk 2]
```diff
@@ -87,7 +83,13 @@ This command triggers the `build` script placed in the **package.json** file of
 
 The `build` script generates **build/index.js**. This file must be uploaded to EAS Build as a part of your project archive, so that the builder can run your function. Ensure that the file is not excluded by a **.gitignore** file or **.easignore** file.
 
-### Connect the function to the custom build config
+</Step>
+
+<Step label="3">
+
+## Connect the function to the custom build config
```
Comment (sjchmiela) [inline docs/pages/custom-builds/functions.mdx:90]:
```suggestion
## Expose the function to the custom build config
```
```suggestion
## Define the function in the custom build config
```
```suggestion
## Import the function in the custom build config
```

I think these are more… better

File: docs/pages/custom-builds/functions.mdx [hunk 3]
```diff
@@ -87,7 +83,13 @@ This command triggers the `build` script placed in the **package.json** file of
...
+
+> **Note:** The following example assumes that you have already set up a custom build workflow and connected it with your **eas.json**. If you haven't, see our ["Get started with custom builds" guide](/custom-builds/get-started/#create-a-workflow) first.
```
Comment (sjchmiela) [inline docs/pages/custom-builds/functions.mdx:92]:
```suggestion
> **Note:** The following example assumes that you have already set up a custom build workflow and configured it in your **eas.json**. If you haven't, see our [“Get started with custom builds” guide](/custom-builds/get-started/#create-a-workflow) first.
```

**Review by sjchmiela — APPROVED** [review]
Looks reasonable to me

**Review by Simek — APPROVED** [review]
👌

File: docs/pages/custom-builds/functions.mdx [hunk 4]
```diff
@@ -87,7 +83,13 @@ This command triggers the `build` script placed in the **package.json** file of
...
 The `build` script generates **build/index.js**. This file must be uploaded to EAS Build as a part of your project archive, so that the builder can run your function. Ensure that the file is not excluded by a **.gitignore** file or **.easignore** file.
 
-### Connect the function to the custom build config
+</Step>
+
+<Step label="3">
+
+## Expose the function to the custom build config
+
+> **Note:** The following example assumes that you have already set up a custom build workflow and configured it in your **eas.json**. If you haven't, see our [“Get started with custom builds” guide](/custom-builds/get-started/#create-a-workflow) first.
```
Comment (amandeepmittal) [inline docs/pages/custom-builds/functions.mdx:92]:
```suggestion
> **Note**: The following example assumes that you have already set up a custom build workflow and configured it in your **eas.json**. If not, see [Get started with custom builds](/custom-builds/get-started/#create-a-workflow) before proceeding.
```

- No quotes in when referring internal link. The link color highlight is enough to grab attention of a user if they don't skim the note.
- Also, we don't have to explicitly call out the type of the document here.

Comment (szdziedzic) [reply docs/pages/custom-builds/functions.mdx]: thanks!

**Review by amandeepmittal — APPROVED** [review]
Thanks for updating this doc and incorporating those changes!

**Pull request merged**